From db1c9a4fe75d7eb971dcfdb356b2c9a645462bfc Mon Sep 17 00:00:00 2001
From: Markus Valentin <valy@systemausfall.org>
Date: Tue, 20 Jun 2017 21:01:31 +0200
Subject: [PATCH 4/8] arm: mxs: read the whole mac address from fuses

before the code assumed a fixed part of the mac address we want to be
hundert percent flexible that why we read the first 32 bits from
hw_cust0 register and the remaining 16 bits from hw_cust1.

Signed-off-by: Markus Valentin <valy@systemausfall.org>
---
 arch/arm/cpu/arm926ejs/mxs/mxs.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/arch/arm/cpu/arm926ejs/mxs/mxs.c b/arch/arm/cpu/arm926ejs/mxs/mxs.c
index 229862079a..56e2389708 100644
--- a/arch/arm/cpu/arm926ejs/mxs/mxs.c
+++ b/arch/arm/cpu/arm926ejs/mxs/mxs.c
@@ -261,12 +261,17 @@ void imx_get_mac_from_fuse(int dev_id, unsigned char *mac)
 	}
 
 	data = readl(&ocotp_regs->hw_ocotp_cust0);
+	mac[0] = (data >> 24 ) & 0xff;
+	mac[1] = (data >> 16) & 0xff;
+	mac[2] = (data >> 8) & 0xff;
+	mac[3] = data & 0xff;
 
-	mac[2] = (data >> 24) & 0xff;
-	mac[3] = (data >> 16) & 0xff;
-	mac[4] = (data >> 8) & 0xff;
-	mac[5] = data & 0xff;
-	mx28_adjust_mac(dev_id, mac);
+	data = readl(&ocotp_regs->hw_ocotp_cust1);
+	mac[4] = (data >>24) & 0xff;
+	mac[5] = (data >>16) & 0xff;
+
+	//we dont call this as we maybe will have multiple mac address ranges
+	//mx28_adjust_mac(dev_id, mac);
 }
 #else
 void imx_get_mac_from_fuse(int dev_id, unsigned char *mac)
-- 
2.11.0

